merge_sort.py

```diff
@@ -31,8 +31,31 @@ def merge_sort(lista_numeros: list[int]):
     >>> merge_sort([1, 3, 1])
     [1, 1, 3]
     """
-
-    # TODO implemente o código fazendo ordenação baseado na ideia do merge sort.
-    # TODO troque a linha abaixo pelo return com a variável que guarda a lista como um todo.
-    # TODO quando terminar de implementar apague esses comentários.
-    return 0
+    def merge_sort(lista_numeros):
+        if len(lista_numeros) > 1:
+            mid = len(lista_numeros)//2
+            L = lista_numeros[:mid]
+            R = lista_numeros[mid:]
+            merge_sort(L)
+            merge_sort(R)
+            i = j = k = 0
+            while i < len(L) and j < len(R):
+                if L[i] < R[j]:
+                    lista_numeros[k] = L[i]
+                    i += 1
+                else:
+                    lista_numeros[k] = R[j]
+                    j += 1
+                k += 1
+            while i < len(L):
+                lista_numeros[k] = L[i]
+                i += 1
+                k += 1
+    
+            while j < len(R):
+                lista_numeros[k] = R[j]
+                j += 1
+                k += 1
+        return lista_numeros
```

Comment:
```suggestion
    if len(lista_numeros) > 1:
        mid = len(lista_numeros)//2
        L = lista_numeros[:mid]
        R = lista_numeros[mid:]
        merge_sort(L)
        merge_sort(R)
        i = j = k = 0
        while i < len(L) and j < len(R):
            if L[i] < R[j]:
                lista_numeros[k] = L[i]
                i += 1
            else:
                lista_numeros[k] = R[j]
                j += 1
            k += 1
        while i < len(L):
            lista_numeros[k] = L[i]
            i += 1
            k += 1

        while j < len(R):
            lista_numeros[k] = R[j]
            j += 1
            k += 1
    return lista_numeros
```
Aqui também tem uma função dentro de outra.